frontend/packages/dev-console/src/components/pipelines/__tests__/PipelineDetailsPage.spec.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import { referenceForModel } from '@console/internal/module/k8s';
+import { DetailsPageProps } from '@console/internal/components/factory';
+import { PipelineModel } from '../../../models';
+import PipelineDetailsPage from '../PipelineDetailsPage';
+
+const pipelineDetailsProps: DetailsPageProps = {
+  match: {
+    isExact: true,
+    path: `/k8s/ns/:ns/${referenceForModel(PipelineModel)}/:name/`,
+    url: 'k8s/ns/tekton-pipelines/tekton.dev~v1alpha1~Pipeline/simple-pipeline',
+    params: {
+      ns: 'tekton-pipelines',
+    },
+  },
+  kind: 'Pipeline',
+};
+
+const expectedTabs: string[] = [`Overview`, `YAML`, `Pipeline Runs`, `Parameters`, `Resources`];
+
+describe('Pipeline details page tests', () => {
+  it('should render `DetailsPage` Component with proper page tabs -> [ `Overview` , `YAML` , `Pipeline Runs`, `Parameters`, `Resources` ]', () => {
+    const pipelineRunDetailsWrapper = shallow(<PipelineDetailsPage {...pipelineDetailsProps} />);
+    const pages = pipelineRunDetailsWrapper.find('DetailsPage').prop('pages');
+
+    expect(pipelineRunDetailsWrapper.find('DetailsPage').exists()).toBe(true);
+    expectedTabs.forEach((tab, i) => {
+      expect(pages[i].name).toBe(tab);
+    });
+  });
```

Comment:
Also add a test to check for `ErrorPage404` component when the `errorCode` is 404

Reply:
this test doesn't exist for other resources.... This is already a fallback. So, this check is unnecessary.

Reply:
I'm inclined to agree with @debsmita1 on this one... if it's in the code, we should test it. 

We need to validate the conditional paths as part of these tests. The only alternative I can think of is to log a refactor task to convert `PipelineDetailsPage` to a functional component and remove the dependency on `ErrorPage404` but even then, it's not likely to be done anytime soon. So I think we should check this logical path as part of these tests.

Reply:
fwiw i would agree.  Need to hit several branches for a test to be solid.